src/QoreJniClassMap.h

```diff
@@ -100,114 +100,118 @@ class QoreJniClassMapBase {
 
 class QoreJniClassMap : public QoreJniClassMapBase {
 public:
-   static QoreJniThreadLock m;
+    static QoreJniThreadLock m;
 
-   DLLLOCAL void init();
+    DLLLOCAL void init();
 
-   DLLLOCAL void destroy(ExceptionSink& xsink);
+    DLLLOCAL void destroy(ExceptionSink& xsink);
 
-   DLLLOCAL QoreValue getValue(LocalReference<jobject>& jobj);
+    DLLLOCAL QoreValue getValue(LocalReference<jobject>& jobj);
 
-   DLLLOCAL const QoreTypeInfo* getQoreType(jclass cls);
+    DLLLOCAL const QoreTypeInfo* getQoreType(jclass cls);
 
-   DLLLOCAL void initDone() {
-      assert(!init_done);
-      init_done = true;
-   }
+    DLLLOCAL void initDone() {
+        assert(!init_done);
+        init_done = true;
+    }
 
-   DLLLOCAL QoreNamespace& getJniNs() {
-      return *default_jns;
-   }
+    DLLLOCAL QoreNamespace& getJniNs() {
+        return *default_jns;
+    }
+
+    DLLLOCAL jobject getJavaObject(const QoreObject* o);
 
-   DLLLOCAL jobject getJavaObject(const QoreObject* o);
+    DLLLOCAL jarray getJavaArray(const QoreListNode* l, jclass cls);
 
-   DLLLOCAL jarray getJavaArray(const QoreListNode* l, jclass cls);
+    DLLLOCAL jclass findLoadClass(const char* name);
+    DLLLOCAL jclass findLoadClass(const QoreString& name);
 
-   DLLLOCAL jclass findLoadClass(const char* name);
-   DLLLOCAL jclass findLoadClass(const QoreString& name);
+    DLLLOCAL QoreBuiltinClass* findCreateQoreClass(LocalReference<jclass>& jc);
+    DLLLOCAL QoreBuiltinClass* findCreateQoreClass(const char* name);
 
-   DLLLOCAL QoreBuiltinClass* findCreateQoreClass(LocalReference<jclass>& jc);
-   DLLLOCAL QoreBuiltinClass* findCreateQoreClass(const char* name);
+    DLLLOCAL QoreBuiltinClass* findCreateQoreClassInProgram(QoreString& name, const char* jpath, Class* c);
```

Comment:
why are we forcing the caller to pass the `name` as `QoreString&` so it can be changed in the function. Is it somehow output parameter?

Reply:
yes, it's an input/output param; I updated the function docs in the header about this